h/static/scripts/group-forms/components/EditGroupMembersForm.tsx

```diff
@@ -49,24 +72,52 @@ export default function EditGroupMembersForm({
     }
     const abort = new AbortController();
     setErrorMessage(null);
-    fetchMembers(config.api.readGroupMembers, abort.signal)
+    fetchMembers(config.api.readGroupMembers, userid, abort.signal)
       .then(setMembers)
       .catch(err => {
         setErrorMessage(`Failed to fetch group members: ${err.message}`);
       });
     return () => {
       abort.abort();
     };
-  }, [config.api.readGroupMembers]);
+  }, [config.api.readGroupMembers, userid]);
 
   const columns = [
     {
       field: 'username' as keyof MemberRow,
       label: 'Username',
     },
+    {
+      field: 'canDelete' as keyof MemberRow,
+      label: '',
+      classes: 'w-[55px]',
+    },
   ];
 
   const [filter, setFilter] = useState('');
+  const [pendingRemoval, setPendingRemoval] = useState<string | null>(null);
+
+  const removeUserFromGroup = async (username: string) => {
+    // istanbul ignore next
+    if (!members || !config.api.removeGroupMember) {
+      return;
+    }
+    const member = members.find(m => m.username === username);
+    // istanbul ignore next
+    if (!member) {
+      return;
+    }
+    setPendingRemoval(null);
+
+    try {
+      await removeMember(config.api.removeGroupMember, member.userid);
+      setMembers(members =>
+        members ? members.filter(m => m.userid !== member.userid) : null,
+      );
```

Comment:
```suggestion
      setMembers(members =>
        members?.filter(m => m.userid !== member.userid),
      );
```

Reply:
This doesn't work at present because `members` is defined as `MemberRow[] | null` so you can't assign `undefined` to it. The code could potentially be simplified by changing the type to `MemberRow[] | undefined` instead.

Reply:
Oh! For some reason I thought `member?.filter()` would resolve to `null` if `members` is `null`, not to `undefined` 😅

Then yeah, it's fine as it is.